packages/oidc-dpop-client-browser/src/cleanup/cleanup.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { OidcClient, WebStorageStateStore } from "oidc-client";
+
+/**
+ * Removes OIDC-specific query parameters from a given URL (state, code...)
+ * @param redirectUrl The URL to clean up.
+ * @returns A copy of the URL, without OIDC-specific query params.
+ */
+export function cleanupRedirectUrl(redirectUrl: string): string {
+  const cleanedUrl = new URL(redirectUrl);
+  cleanedUrl.searchParams.delete("code");
+  cleanedUrl.searchParams.delete("state");
+  return cleanedUrl.toString();
+}
+
+/**
+ * Clears any OIDC-related data lingering in the local storage.
+ */
+export async function clearOidcPersistentStorage(): Promise<void> {
```

Comment:
This is specifically referring to `oidc-client-js`, right? If so, I'd try and make that a little bit more specific:
```suggestion
export async function clearOidcClientPersistentStorage(): Promise<void> {
```
Regardless, this seems like a very heavy-handed, and a very 'implicit' thing to be doing - i.e instantiating a new instance of an object just to call a 'clear state' method on that instance (i.e. we both know how `oidc-client-js` works, but this behaviour seems very counter-intuitive to me). But anyway, this whole PR requires more serious consideration to fix up 'properly' later I think (i.e. an `OidcClient` instance per session maybe, or just using their `UserManager` from the start...).

Reply:
Internally, yes, it is specific to `oidc-client-js`, but since this function is exposed as part of the API of `oidc-dpop-client-js` I guessed it wasn't needed to be specific about the underlying library.

Moreover, I think their approach is motivated by the fact that in the browser, it's impossible to have one `OidcClient` instance per session, since there is a redirection in the OIDC flow that necessarily will lead to the page being reloaded. They must assume that there is only one session active at a time, and their architecture is based on this assumption.